### ISSUE TITLE
Magento 2.0.7 XML sitemap is not generated by schedule

### ISSUE THREAD

**Issue by dexterityzx** [magento/magento2#5768]
### Preconditions
1.  Magento 2.0.7
2.  Cron job is working. I've checked the cron_schedule table, but nothing like "sitemap" is in the table.
### Steps to reproduce
1. go to admin>store>configurtion>catalog>xml sitemap
2. turn on generation and set up time.
### Expected result
1.  generate sitemap in magento root folder.
### Actual result

no xml sitemap is found. Below is my settings.
1. Cron settings:

```
*/1 * * * * php /var/www/magento-demo1/bin/magento cron:run | grep -v "Ran jobs by schedule" >> /var/www/magento-demo1/var/log/magento.cron.log&
*/1 * * * * php /var/www/magento-demo1/update/cron.php >> /var/www/magento-demo1/var/log/update.cron.log&
*/1 * * * * php /var/www/magento-demo1/bin/magento setup:cron:run >> /var/www/magento-demo1/var/log/setup.cron.log&
```
1. Log: magento.cron.log and setup.cron.log are empty, and the update.cron.log shows _Cron readiness check failed_
2. cron_schedule table:
   ![capture](https://cloud.githubusercontent.com/assets/13892193/17067678/d4b6c430-501a-11e6-95ea-fb1a07723c40.PNG)

**Comment by magento-engcom-team** [comment]
@dexterityzx, thank you for your report.
We've created internal ticket(s) MAGETWO-80284 to track progress on the issue.

**Comment by mrodespin** [comment]
Can anyone assign me this ticket?

**Comment by maksek** [comment]
@mrodespin assigned.

**Comment by jameshalsall** [comment]
@dexterityzx if you configure a sitemap generation time in `Stores > Configuration > Catalog > XML Sitemap > Generation Settings`, and then install https://github.com/netz98/n98-magerun2 and run

```
n98-magerun2.phar sys:cron:list
```

do you see a `sitemap_generate` job for the correct time?

**Comment by magento-engcom-team** [comment]
Hi @dexterityzx. Thank you for your report. 
The issue has been fixed in magento/magento2#14822 by @jameshalsall in 2.2-develop branch
Related commit(s):
 - [9463c7768578d69d5919cf78276f686139b3282b](https://github.com/magento/magento2/commit/9463c7768578d69d5919cf78276f686139b3282b)
 - [d95f0449bcdb247f5e2f2352707a875f64f50272](https://github.com/magento/magento2/commit/d95f0449bcdb247f5e2f2352707a875f64f50272)
 - [a065434d9c55e3570ab369a692565d1af514deec](https://github.com/magento/magento2/commit/a065434d9c55e3570ab369a692565d1af514deec)

The fix will be available with the upcoming 2.2.5 release.

**Comment by VladimirZaets** [comment]
Hi @dexterityzx. Thank you for your report. 
The issue has been fixed in magento/magento2#14908 by @rostyslav-hymon in 2.3-develop branch
Related commit(s):
 - [0b859217153c9b1ef97a0392166d76272d5a2ac8](https://github.com/magento/magento2/commit/0b859217153c9b1ef97a0392166d76272d5a2ac8)
 - [01f6fc21a691923663faafdc5e8e1df1073d3f23](https://github.com/magento/magento2/commit/01f6fc21a691923663faafdc5e8e1df1073d3f23)
 - [a324f86feb372a52b6b8bacf470639c8fd6d77a9](https://github.com/magento/magento2/commit/a324f86feb372a52b6b8bacf470639c8fd6d77a9)

The fix will be available with the upcoming 2.3.0 release.

**Comment by magento-engcom-team** [comment]
Hi @dexterityzx. Thank you for your report. 
The issue has been fixed in magento/magento2#15159 by @rogyar in 2.1-develop branch
Related commit(s):
 - [f4ae0e32abc1de2cfc5db23260df0e82ee68f75a](https://github.com/magento/magento2/commit/f4ae0e32abc1de2cfc5db23260df0e82ee68f75a)

The fix will be available with the upcoming 2.1.15 release.
